packages/x-date-pickers/src/YearPicker/PickersYear.tsx

```diff
@@ -141,23 +145,27 @@ export const PickersYear = React.forwardRef<HTMLButtonElement, YearProps>(functi
       data-mui-test="year"
       className={clsx(classes.root, className)}
       ownerState={ownerState}
+      {...other}
```

Comment:
Similar issue is also happening here because of the change where `other` is spread.
![Screenshot 2022-09-06 at 16 59 26](https://user-images.githubusercontent.com/4941090/188655491-c090a87b-7962-4b32-96ea-aa8432c0aac7.png)
`aria-current` should be on the `button` instead of the wrapping `div`.

Reply:
Indeed, I may have been a bit quick on my change.
Typescript was not complaining that `aria-current` was not present in the props.

I checked and unless I missed one, all the other props passed to `PickersYear` and `PickersMonth` are correctly handled.

Reply:
Yes, as far as I've tested other places with generic examples, it seemed to work fine. 👌